sdk/cloudmachine/Azure.Provisioning.CloudMachine/src/FeaturesExtensions/AIFoundry.cs

```diff
@@ -0,0 +1,185 @@
+                )
+            )
+        );
+        return [cs];
+    }
+}
+
+internal class AIFoundryHubCdk : NamedProvisionableConstruct
+{
+    public AIFoundryHubCdk(string bicepIdentifier, string name)
+        : base(bicepIdentifier)
+    {
+        Name = name;
+        FriendlyName = name;
+    }
+
+    public string Name { get; set; }
+
+    public string FriendlyName { get; set; }
+
+    protected override IEnumerable<BicepStatement> Compile()
+    {
+        ResourceStatement hub = new(
+            BicepIdentifier,
+            new StringLiteralExpression("Microsoft.MachineLearningServices/workspaces@2023-08-01-preview"),
+                new ObjectExpression(
+                new PropertyExpression("name", Name),
+                new PropertyExpression("location", new IdentifierExpression("location")),
+                new PropertyExpression("kind", "hub"),
+                new PropertyExpression("identity",
+                    new ObjectExpression(
+                        new PropertyExpression("type", "SystemAssigned")
+                    )
+                ),
+                new PropertyExpression("properties",
+                    new ObjectExpression(
+                        new PropertyExpression("friendlyName", FriendlyName),
+                        new PropertyExpression("publicNetworkAccess", "Enabled")
+                        //new PropertyExpression("keyVault", ...),
+                        //new PropertyExpression("storageAccount", ...)
+                    )
+                )
+            )
+        );
+        return [hub];
+    }
+}
+
+internal class AIFoundryProjectCdk : ProvisionableResource
+{
+    public AIFoundryProjectCdk(string bicepIdentifier, string name, AIFoundryHubCdk hub)
+        : base(bicepIdentifier, new Core.ResourceType("Microsoft.MachineLearningServices/workspaces"), "2023-08-01-preview")
+    {
+        Name = name;
+        Hub = hub;
+        FriendlyName = name;
+    }
+
+    public string Name { get; set; }
+    public AIFoundryHubCdk Hub { get; set; }
+
+    public string FriendlyName { get; set; }
+
+    public List<AIFoundryConnectionCdk> Connections { get; } = new List<AIFoundryConnectionCdk>();
+    protected override IEnumerable<BicepStatement> Compile()
+    {
+        List<ResourceStatement> resources = new();
+
+        var hubId = new MemberExpression(new IdentifierExpression(Hub.BicepIdentifier), "id");
+
+        ResourceStatement project = new(
+            BicepIdentifier,
+            new StringLiteralExpression($"{base.ResourceType}@{base.ResourceVersion}"),
+                new ObjectExpression(
+                new PropertyExpression("name", Name),
+                new PropertyExpression("location", new IdentifierExpression("location")),
+                new PropertyExpression("kind", "Project"),
+                new PropertyExpression("identity",
+                    new ObjectExpression(
+                        new PropertyExpression("type", "SystemAssigned")
```

Comment:
can we utilize our user-assigned CM identity here?

Reply:
we decided to do as a separate PR